programs/Cargo.toml

```diff
@@ -0,0 +1,35 @@
+[package]
+name = "solana-programs"
```

Comment:
call this “solana-core-programs” perhaps, and move it into “core-programs/“

Reply:
Don't think that would resolve dependency bit i think i can remove runtime's dep on programs, will give that a shot

Reply:
Yeah that comment was just about cleaning up the name.  It’s weird to just call it “solana-programs” imo, that’s too general of a name

Reply:
This crate is used by more than just core, What do you think about "default-programs" at "/default-programs/".

Reply:
Are all the dependencies (aside from runtime/) downstream of `core/`?  if so maybe this turns into `core/src/default_programs.rs`?   I'm not seeing a lot of value having this as a separate crate right now

Reply:
/genesis also depends on it

Reply:
🤔 I don't like "default-programs" cause it means different things to different parts of the tree.  The default programs in runtime/ are not the same as the default programs in genesis/ (or local_cluster).

This naming convention like this clearly positions the programs next to the crate that primarily depends on them:
* `runtime-programs/`
* `core-programs/` (yeah genesis/ depends on core-programs/ too, but the "stronger" dependency is from core/)

Also an instruction_processor is not a thing, ITYM program.  Until we settle on the final name for program, using two names for the one thing just makes our world more confusing!